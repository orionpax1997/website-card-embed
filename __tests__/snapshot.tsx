import { render } from '@testing-library/react';
import WebsiteCardIntroduction from '@/components/website-card-introduction';
import WebsiteCardIntroductionMobile from '@/components/website-card-introduction-mobile';
import WebsiteCardIntroductionLoading from '@/components/website-card-introduction-loading';
import WebsiteCardIntroductionError from '@/components/website-card-introduction-error';

describe('snapshot', () => {
  it('build WebsiteCardIntroduction', () => {
    const analysisData = {
      url: 'https://blog.humblex.top/',
      title: '向之礼的个人博客',
      description: '一个打工人的个人博客小站',
      image: 'https://images.weserv.nl/?url=https://blog.humblex.top/images/avatar.jpg',
      favicon:
        'https://t3.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&fallback_opts=TYPE,SIZE,URL&url=https://blog.humblex.top/&size=16',
    };
    const { container } = render(<WebsiteCardIntroduction {...analysisData} />);
    expect(container).toMatchSnapshot();
  });

  it('build WebsiteCardIntroductionMobile', () => {
    const analysisData = {
      url: 'https://blog.humblex.top/',
      title: '向之礼的个人博客',
      description: '一个打工人的个人博客小站',
      image: 'https://images.weserv.nl/?url=https://blog.humblex.top/images/avatar.jpg',
      favicon:
        'https://t3.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&fallback_opts=TYPE,SIZE,URL&url=https://blog.humblex.top/&size=16',
    };
    const { container } = render(<WebsiteCardIntroductionMobile {...analysisData} />);
    expect(container).toMatchSnapshot();
  });

  it('build WebsiteCardIntroductionLoading', () => {
    const { container } = render(<WebsiteCardIntroductionLoading hasImage={true} />);
    expect(container).toMatchSnapshot();
  });

  it('build WebsiteCardIntroductionError', () => {
    const { container } = render(<WebsiteCardIntroductionError url="https://blog.humblex.top/" />);
    expect(container).toMatchSnapshot();
  });
});
