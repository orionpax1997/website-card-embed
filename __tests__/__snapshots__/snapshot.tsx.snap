// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot build WebsiteCardIntroduction 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  position: relative;
  width: 100%;
  max-width: 1200px;
  min-width: 620px;
  border-width: 1px;
  border-radius: lg;
}

.emotion-1 a[href]:not(.chakra-linkbox__overlay),
.emotion-1 abbr[title],
.emotion-1 button {
  position: relative;
  z-index: 1;
}

.emotion-1:hover,
.emotion-1[data-hover] {
  background-color: rgba(55, 53, 47, 0.08);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 120px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 60%;
  padding: 15px;
}

.emotion-3>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-4 {
  font-size: sm;
  font-weight: semibold;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-5 {
  position: static;
}

.emotion-5::before {
  content: '';
  cursor: inherit;
  display: block;
  position: absolute;
  top: 0px;
  left: 0px;
  z-index: 1;
  width: 100%;
  height: 100%;
}

.emotion-7 {
  font-size: xs;
  color: gray.500;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: var(--chakra-line-clamp);
  --chakra-line-clamp: 2;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-8>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 10px;
  margin-inline-start: 10px;
}

.emotion-9 {
  width: 16px;
  height: 16px;
  border-radius: sm;
}

.emotion-10 {
  font-size: xs;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-11 {
  object-fit: cover;
  width: 40%;
  height: 100%;
  border-radius: sm;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="chakra-linkbox emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="chakra-stack emotion-3"
        >
          <h4
            class="emotion-4"
          >
            <a
              class="chakra-linkbox__overlay emotion-5"
              href="https://humble-blog.vercel.app/"
              target="_blank"
            >
              向之礼的个人博客
            </a>
          </h4>
          <div
            class="emotion-6"
          >
            <p
              class="chakra-text emotion-7"
            >
              一个打工人的个人博客小站
            </p>
          </div>
          <div
            class="chakra-stack emotion-8"
          >
            <img
              alt="website card icon"
              class="chakra-image emotion-9"
              src="https://t3.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&fallback_opts=TYPE,SIZE,URL&url=https://humble-blog.vercel.app/&size=16"
            />
            <p
              class="chakra-text emotion-10"
            >
              https://humble-blog.vercel.app/
            </p>
          </div>
        </div>
        <img
          alt="website card image"
          class="chakra-image emotion-11"
          src="https://images.weserv.nl/?url=https://humble-blog.vercel.app/images/avatar.jpg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot build WebsiteCardIntroductionError 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  position: relative;
  width: 100%;
  max-width: 1200px;
  min-width: 600px;
  border-width: 1px;
  border-radius: lg;
}

.emotion-1 a[href]:not(.chakra-linkbox__overlay),
.emotion-1 abbr[title],
.emotion-1 button {
  position: relative;
  z-index: 1;
}

.emotion-1:hover,
.emotion-1[data-hover] {
  background-color: rgba(55, 53, 47, 0.08);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 120px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  padding: 15px;
}

.emotion-3>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-4 {
  font-size: sm;
  font-weight: semibold;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-5 {
  position: static;
}

.emotion-5::before {
  content: '';
  cursor: inherit;
  display: block;
  position: absolute;
  top: 0px;
  left: 0px;
  z-index: 1;
  width: 100%;
  height: 100%;
}

.emotion-7 {
  font-size: xs;
  color: gray.500;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: var(--chakra-line-clamp);
  --chakra-line-clamp: 2;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-8>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 10px;
  margin-inline-start: 10px;
}

.emotion-9 {
  font-size: xs;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="chakra-linkbox emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="chakra-stack emotion-3"
        >
          <h4
            class="emotion-4"
          >
            <a
              class="chakra-linkbox__overlay emotion-5"
              href="https://humble-blog.vercel.app/"
              target="_blank"
            >
              很抱歉, 嵌入网址的描述信息无法正常解析
            </a>
          </h4>
          <div
            class="emotion-6"
          >
            <p
              class="chakra-text emotion-7"
            >
              请直接点击卡片访问原网页查看。或者您可以补充更多参数, 来跳过解析直接展示静态卡片。
            </p>
          </div>
          <div
            class="chakra-stack emotion-8"
          >
            <p
              class="chakra-text emotion-9"
            >
              https://humble-blog.vercel.app/
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`snapshot build WebsiteCardIntroductionLoading 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 120px;
  max-width: 1200px;
  border-width: 1px;
  border-radius: lg;
  width: 100%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 60%;
  padding: 15px;
}

.emotion-2>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-3 {
  box-shadow: none;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  cursor: default;
  color: transparent;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 20px;
}

.emotion-3::before,
.emotion-3::after,
.emotion-3 * {
  visibility: hidden;
}

.emotion-4 {
  box-shadow: none;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  cursor: default;
  color: transparent;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 40px;
}

.emotion-4::before,
.emotion-4::after,
.emotion-4 * {
  visibility: hidden;
}

.emotion-5 {
  box-shadow: none;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  cursor: default;
  color: transparent;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  height: 16px;
}

.emotion-5::before,
.emotion-5::after,
.emotion-5 * {
  visibility: hidden;
}

.emotion-6 {
  box-shadow: none;
  -webkit-background-clip: padding-box;
  background-clip: padding-box;
  cursor: default;
  color: transparent;
  pointer-events: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 40%;
  height: 100%;
}

.emotion-6::before,
.emotion-6::after,
.emotion-6 * {
  visibility: hidden;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="chakra-stack emotion-2"
      >
        <div
          class="chakra-skeleton emotion-3"
        />
        <div
          class="chakra-skeleton emotion-4"
        />
        <div
          class="chakra-skeleton emotion-5"
        />
      </div>
      <div
        class="chakra-skeleton emotion-6"
      />
    </div>
  </div>
</div>
`;

exports[`snapshot build WebsiteCardIntroductionMobile 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  position: relative;
  width: 100%;
  max-width: 620px;
  border-width: 1px;
  border-radius: lg;
}

.emotion-1 a[href]:not(.chakra-linkbox__overlay),
.emotion-1 abbr[title],
.emotion-1 button {
  position: relative;
  z-index: 1;
}

.emotion-1:hover,
.emotion-1[data-hover] {
  background-color: rgba(55, 53, 47, 0.08);
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 15px;
}

.emotion-2>*:not(style)~*:not(style) {
  margin-top: 0.5rem;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 0px;
  margin-inline-start: 0px;
}

.emotion-3 {
  font-size: sm;
  font-weight: semibold;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-4 {
  position: static;
}

.emotion-4::before {
  content: '';
  cursor: inherit;
  display: block;
  position: absolute;
  top: 0px;
  left: 0px;
  z-index: 1;
  width: 100%;
  height: 100%;
}

.emotion-6 {
  font-size: xs;
  color: gray.500;
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: var(--chakra-line-clamp);
  --chakra-line-clamp: 2;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-7>*:not(style)~*:not(style) {
  margin-top: 0px;
  -webkit-margin-end: 0px;
  margin-inline-end: 0px;
  margin-bottom: 0px;
  -webkit-margin-start: 10px;
  margin-inline-start: 10px;
}

.emotion-8 {
  width: 16px;
  height: 16px;
  border-radius: sm;
}

.emotion-9 {
  font-size: xs;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="chakra-linkbox emotion-1"
    >
      <div
        class="chakra-stack emotion-2"
      >
        <h4
          class="emotion-3"
        >
          <a
            class="chakra-linkbox__overlay emotion-4"
            href="https://humble-blog.vercel.app/"
            target="_blank"
          >
            向之礼的个人博客
          </a>
        </h4>
        <div
          class="emotion-5"
        >
          <p
            class="chakra-text emotion-6"
          >
            一个打工人的个人博客小站
          </p>
        </div>
        <div
          class="chakra-stack emotion-7"
        >
          <img
            alt="website card icon"
            class="chakra-image emotion-8"
            src="https://t3.gstatic.com/faviconV2?client=SOCIAL&type=FAVICON&fallback_opts=TYPE,SIZE,URL&url=https://humble-blog.vercel.app/&size=16"
          />
          <p
            class="chakra-text emotion-9"
          >
            https://humble-blog.vercel.app/
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
